metron-platform/metron-pcap-backend/src/main/java/org/apache/metron/spout/pcap/HDFSWriterCallback.java

```diff
@@ -96,16 +109,18 @@ public HDFSWriterCallback withConfig(HDFSWriterConfig config) {
         this.config = config;
         return this;
     }
+
     @Override
     public List<Object> apply(List<Object> tuple, EmitContext context) {
-
-        List<Object> keyValue = (List<Object>) tuple.get(0);
-        LongWritable ts = (LongWritable) keyValue.get(0);
-        BytesWritable rawPacket = (BytesWritable)keyValue.get(1);
+        byte[] key = (byte[]) tuple.get(0);
+        byte[] value = (byte[]) tuple.get(1);
+        if(!config.getDeserializer().deserializeKeyValue(key, value, KeyValue.key.get(), KeyValue.value.get())) {
+            LOG.debug("Dropping malformed packet...");
```

Comment:
I'm good with that. I had mostly discarded the worry about size because this is in debugging statements anyway and typically with storm you're setting reasonable timeouts on logging levels anyway.